clang/include/clang/Basic/DiagnosticParseKinds.td

```diff
@@ -783,6 +783,9 @@ def err_ms_property_expected_comma_or_rparen : Error<
 def err_ms_property_initializer : Error<
   "property declaration cannot have a default member initializer">;
 
+def err_assume_attr_expects_cond_expr : Error<
+  "use of this expression in an 'assume' attribute requires parentheses">;
```

Comment:
Since `assume` (currently) only has one spelling (because `clang::assume` is something else as I understand it), I’ve inlined it here. Should I change this to `%0` and pass the attribute name to the diagnostic instead? The same also applies to some of the other assume-related diagnostics

Reply:
Speaking of `clang::assume`: We could add a warning in Sema that suggests using that instead if someone writes e.g. `[[assume("foo")]]` seeing as that tautological, albeit valid.

Reply:
Please pass the name, it will allow us to better handle other spellings (since we allow standards spellings to work with other spellings, which I think `__attribute__((assume))` would work).